### PR TITLE
Change `clean.ps1` (and `clean.cmd`) to restore deleted files after cleaning

### PR DESCRIPTION
**Pull request by dougbu**
- #6845

### CODE REVIEW

File: clean.ps1 [hunk 1]
```diff
@@ -38,4 +38,5 @@ if ($Help) {
     exit 0
 }
 
-git clean -dix -e .dotnet/ -e .tools/ -e src/SignalR/clients/ts/FunctionalTests/node_modules/ @GitArguments
+git clean -dix -e .dotnet/ -e .tools/ @GitArguments
+git checkout -- $(git ls-files -d)
```
Comment (Tratcher) [inline clean.ps1]: How is this different from `git reset --hard`?

Comment (dougbu) [reply clean.ps1]: It only restores deleted files where the deletion hasn't been staged. It also doesn't undo modifications to checked-in files, unstage anything, and so on.

**Comment by dougbu** [comment]
Bump. Could someone please approve?

**Review by Tratcher — APPROVED** [review]
Verified locally.

**Review by JunTaoLuo — APPROVED** [review]

**Pull request merged**